### PR TITLE
Refactor Riak CS role

### PR DESCRIPTION
**Pull request by millerdev**
Riak CS can now be deployed to a stack with a single idempotent command:

```
$ ansible-playbook ... deploy_riakcs.yml

# then update localsettings
$ ansible-playbook ... deploy_stack.yml --tags=localsettings
```

Adding a new node to the cluster:
- add host to the `[riakcs]` group in the environment file
- run `ansible-playbook deploy_riakcs.yml`

@snopoke @benrudolph

### CODE REVIEW

File: ansible/group_vars/all.yml [hunk 1]
```diff
@@ -22,9 +22,18 @@ dev_group: dev
...
+riakcs_is_enabled: "{{ 'true' if groups['riakcs'] else '' }}"
+riakcs_control: "{{ groups['riakcs'][0] | default('riakcs-is-disabled') }}"
 riak_port: 8087
+riak_data_dir: "{{ encrypted_root }}/riak"
 stanchion_port: 8085
+stanchion_data_dir: "{{ encrypted_root }}/stanchion"
+
+# commcare-hq connects to an S3-compatible service, which is Riak CS
+s3_blob_db_enabled: "{{ groups['riakcs'] | bool }}"
+s3_blob_db_url: "http://{{ groups['proxy'][0] }}:{{ riakcs_port }}"
```
Comment (snopoke) [inline ansible/group_vars/all.yml]: At some point we should loadbalance between the nodes: http://docs.basho.com/riakcs/latest/cookbooks/configuration/Load-Balancing-and-Proxy-Configuration/#Nginx

File: ansible/roles/riakcs/install/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+riak_version: 2.1.1
+riak_package_release: 1
+riak_package: riak={{ riak_version }}-{{ riak_package_release }}
+riak_script_url: https://packagecloud.io/install/repositories/basho/riak/script.deb.sh
+riak_conf: /etc/riak/riak.conf
+
+riakcs_version: 2.0.1
```
Comment (snopoke) [inline ansible/roles/riakcs/install/defaults/main.yml]: `2.1.0`

File: ansible/roles/riakcs/install/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+riakcs_package_release: 1
+riakcs_package: riak-cs={{ riakcs_version }}-{{ riakcs_package_release }}
+riakcs_script_url: https://packagecloud.io/install/repositories/basho/riak-cs/script.deb.sh
+riakcs_conf: /etc/riak-cs/riak-cs.conf
+
+stanchion_version: 2.0.0
```
Comment (snopoke) [inline ansible/roles/riakcs/install/defaults/main.yml]: `2.1.0`
http://docs.basho.com/riakcs/latest/cookbooks/installing/Installing-Riak-CS/#Version-Compatibility

File: ansible/deploy_riakcs.yml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    - role: riakcs/configure
+      name: riak
+      extra_conf: advanced.config
+      data_dirs:
+        - "{{ riak_data_dir }}"
+        - "{{ riak_ring_dir }}"
+        - "{{ riak_data_root_leveldb }}"
+        - "{{ riak_data_root_bitcask }}"
+
+- name: Join Riak nodes to cluster
```
Comment (snopoke) [inline ansible/deploy_riakcs.yml]: Riak [CS]

File: ansible/group_vars/all.yml (hunk 1 above)
Comment (snopoke) [inline ansible/group_vars/all.yml]: ha, I take that back - misread the line

File: ansible/roles/riakcs/create_admin_user/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+- name: Wait for Riak service to become available
+  command: riak-admin wait-for-service riak_kv {{ riak_nodename }}
+  changed_when: false
+
+# Random email address because a given email can only be used once. It
+# may be necessary to generate a new admin user if something fails
+# between when the new user is created and when its keys are written to
+# the config file (in that case there may be an orphaned admin user
+# for which we do not know the key or secret... do we care?).
+- name: Generate random email for new Riak CS admin user
```
Comment (snopoke) [inline ansible/roles/riakcs/create_admin_user/tasks/main.yml]: did this happen during your testing?

Comment (millerdev) [inline ansible/roles/riakcs/create_admin_user/tasks/main.yml]: yes

File: ansible/roles/riakcs/install/defaults/main.yml (hunk 3 above)
Comment (millerdev) [inline ansible/roles/riakcs/install/defaults/main.yml]: Good catch, and the link was helpful

**Comment by benrudolph** [comment]
really nice! :muscle:

**Comment by millerdev** [comment]
working on the test failure...

**Comment by millerdev** [comment]
:white_check_mark:

File: ansible/roles/riakcs/create_admin_user/tasks/main.yml (hunk 5 above)
Comment (benrudolph) [inline ansible/roles/riakcs/create_admin_user/tasks/main.yml]: haha i used to update it manually! random is definitely the way to go :smile:

**Pull request merged**